### DISCUSSION TITLE
how to obtain number of seats available programatically

### DISCUSSION THREAD

**Discussion by rojas-hmhco** [community/community#63938] [Enterprise] [answered]
### Select Topic Area

Question

### Body

Subject is self explanatory. We are trying to add a metric to datadog that tracks the number of free seats for github enterprise licenses. As an adition, maybe get a list of dormant users as well ( or at least the number of dormant users ).

**Comment by Jaber-Saed** [comment]
To track the number of free seats for GitHub Enterprise licenses and gather metrics related to dormant users using Datadog, you'll need to perform several steps:

    GitHub Enterprise Integration:
    First, set up the integration between GitHub Enterprise and Datadog. This involves connecting your GitHub Enterprise account to Datadog to pull in relevant data. You'll likely need administrative access to configure this.

    Metrics Collection:
    Once the integration is set up, you can use Datadog to collect metrics related to your GitHub Enterprise instance. Metrics can be collected using various methods, such as API calls or agent-based integrations. You might need to utilize GitHub's APIs to retrieve information like the number of free seats and the list of dormant users.

    Custom Metrics:
    To track the number of free seats for GitHub Enterprise licenses, you might need to create a custom metric in Datadog. Depending on the data available through GitHub's APIs, you might need to write scripts or use tools like cURL or Datadog's API to fetch the relevant information and push it as custom metrics.

    Dormant User Metrics:
    Similarly, for tracking dormant users, you'll need to identify what criteria define a user as "dormant." Once you have the criteria, you can create a script that fetches the relevant user data from GitHub's API and calculates the number of dormant users. This information can then be sent to Datadog as custom metrics.

    Alerts and Dashboards:
    Once you have the metrics in Datadog, you can set up alerts to be notified when certain thresholds are reached, such as a specific number of free seats remaining or a certain number of dormant users. Additionally, you can create custom dashboards in Datadog to visualize these metrics for easy monitoring.

    Automation:
    Consider automating the process of fetching data from GitHub's API and sending it to Datadog. This can be done using scripts or tools like Datadog's integrations.

    Periodic Updates:
    Since this information can change frequently, schedule the metrics collection and updates to occur periodically. This ensures that your Datadog metrics remain up to date.

    Documentation and Collaboration:
    Document the process you follow and the scripts/tools you use to set up this monitoring. This documentation will be helpful for other team members and for troubleshooting in the future.

Remember that the specifics of implementation depend on the APIs provided by GitHub Enterprise and the capabilities of Datadog at the time you're implementing this solution. Always refer to the latest documentation and resources for both platforms for the most accurate and up-to-date information.

**Reply by rojas-hmhco** [reply]
Thanks for this information. I hate to be a bother but do you happen to have a link to the documentation for this process? I was looking in the documentation for GitHub Enterprise and could not find how to do this.

**Comment by byrneh** [comment, marked as answer]
Hi @rojas-hmhco the REST API 
https://docs.github.com/en/enterprise-cloud@latest/rest/enterprise-admin/license?apiVersion=2022-11-28#list-enterprise-consumed-licenses

There are also APIs for querying various billing information.

For Dormant users as defined and.calculated by GitHubs Dormancy reports. This is.not provided by an API as you request in browser GUI and a background job is created to.calculate and hie long it takes is.depensant on membership, size and use of your enterprise account

**Comment by rojas-hmhco** [comment]
Thank youu @byrneh  and @Jaber-Saed
